source/common/config/utility.h

```diff
@@ -208,13 +208,12 @@ class Utility {
     const auto transport_api_version = api_config_source.transport_api_version();
     ASSERT_IS_MAIN_OR_TEST_THREAD();
     if (transport_api_version != envoy::config::core::v3::ApiVersion::V3) {
-      const ApiVersion version = ApiVersionInfo::apiVersion();
       const std::string& warning = fmt::format(
           "V2 (and AUTO) xDS transport protocol versions are deprecated in {}. "
           "The v2 xDS major version has been removed and is no longer supported. "
           "You may be missing explicit V3 configuration of the transport API version, "
-          "see the advice in https://www.envoyproxy.io/docs/envoy/v{}.{}.{}/faq/api/envoy_v3.",
-          api_config_source.DebugString(), version.major, version.minor, version.patch);
```

Comment:
@agrawroh im wondering how this is broken

if i browse for example to https://www.envoyproxy.io/docs/envoy/v1.24.0/faq/api/envoy_v3 it works as expected - i think we want the message to point to the correct version of the docs

Reply:
@phlax For some reason it was pointing to `v3.0.0` when I was using v1.24 and trying to add the ExtAuthZ gRPC filter without setting `transport_api_version`.  

```
2022/10/18 20:17:48 WARN DockerService[Thread-4] : envoy-a371437ef2b74340b34a4431c13bb821 - }
2022/10/18 20:17:48 WARN DockerService[Thread-4] : envoy-a371437ef2b74340b34a4431c13bb821 - . The v2 xDS major version has been removed and is no longer supported. You may be missing explicit V3 configuration of the transport API version, see the advice in https://www.envoyproxy.io/docs/envoy/v3.0.0/faq/api/envoy_v3.
```

Reply:
hmm, yep i see its pointing to the api version, not the Envoy version

@wbpcode any ideas on how to fix that ?

Reply:
cc @phlax IMO, simple `latest` is good enough. 
And the `BUILD_VERSION_NUMBER` would be a althernative. But if it's end with `-dev`, we need to calculate the suitable version first.

Reply:
not sure i follow exactly - if it ends with `-dev` then im happy with it just going to `latest` docs

if the binary is (release) versioned then im more concerned it points to the correct version of the docs

we get less now, but historically we got a lot of tickets where there was a mismatch between running version and the docs the user was reading

Reply:
ok - lets just go with `latest` on the basis that its better than what we have now

i think we will probably want to revisit this at some point, as the other issue is that there is no guarantee the page exists in `latest` only in the current branch - i have removed/fixed similar links for this reason in the past